packages/framework/src/Component/Image/Processing/ImageProcessor.php

```diff
@@ -71,30 +72,27 @@ public function createInterventionImage($filepath)
      * @param string $filepath
      * @return string
      */
-    public function convertToShopFormatAndGetNewFilename($filepath)
+    public function convertToShopFormatAndGetNewFilename(string $filepath): string
     {
         $filename = pathinfo($filepath, PATHINFO_FILENAME);
         $extension = strtolower(pathinfo($filepath, PATHINFO_EXTENSION));
         $newFilepath = pathinfo($filepath, PATHINFO_DIRNAME) . '/' . $filename . '.';
 
-        if ($extension === self::EXTENSION_PNG) {
-            $extension = self::EXTENSION_PNG;
-        } elseif (in_array($extension, $this->supportedImageExtensions, true)) {
-            $extension = self::EXTENSION_JPG;
-        } else {
+        if (!in_array($extension, $this->supportedImageExtensions, true)) {
             throw new FileIsNotSupportedImageException($filepath);
         }
-        $newFilepath .= $extension;
-
-        $image = $this->createInterventionImage($filepath);
-        $data = $image->encode($extension)->getEncoded();
 
-        $this->filesystem->delete($filepath);
-        $this->filesystem->write($newFilepath, $data);
+        $newFilepath .= $extension;
```

Comment:
This changes the application's behavior for GIF (and possibly other formats other than png/jpg). 
Previously, the GIF was transformed to the JPG format. 
Now in administration, it stays as a GIF (which means the image can be animated in administration) and is processed by imgproxy, resulting in a still image. 


https://github.com/shopsys/shopsys/assets/1177414/3eff8158-cf21-4229-b695-8c0bc83166f4